integration-tests/src/test/java/org/apache/fineract/integrationtests/LoanProductWithAdvancedPaymentAllocationIntegrationTests.java

```diff
@@ -364,6 +364,42 @@ public void testCreateCumulativeLoanProductWithInterestRefund() {
                 loanProductError.get(0).get("userMessageGlobalisationCode"));
     }
 
+    @Test
+    public void testCreateShouldFailWhenNoNumberOfRepaymentsIsProvided() {
+        // given
+        ResponseSpecification errorResponse = new ResponseSpecBuilder().expectStatusCode(400).build();
+        LoanTransactionHelper validationErrorHelper = new LoanTransactionHelper(REQUEST_SPEC, errorResponse);
+
+        String loanProduct = new LoanProductTestBuilder().withPrincipal("15,000.00").withNumberOfRepayments(null)
+                .withRepaymentAfterEvery("1").withRepaymentTypeAsMonth().withinterestRatePerPeriod("1")
+                .withAccountingRulePeriodicAccrual(new Account[] { ASSET_ACCOUNT, EXPENSE_ACCOUNT, INCOME_ACCOUNT, OVERPAYMENT_ACCOUNT })
+                .withInterestRateFrequencyTypeAsMonths().withAmortizationTypeAsEqualInstallments().withInterestTypeAsDecliningBalance()
+                .withFeeAndPenaltyAssetAccount(FEE_PENALTY_ACCOUNT).withRepaymentStrategy("advanced-payment-allocation-strategy").build();
+
+        // when
+        List<Map<String, String>> loanProductError = validationErrorHelper.getLoanProductError(loanProduct, "errors");
```

Comment:
Wrong exception message: There should be an error which says: The parameter `numberOfRepayments` is mandatory...

Reply:
Done! Updated